src/sai/mod.rs

```diff
@@ -122,34 +142,99 @@ macro_rules! sai_hal {
                     match event {
-                        Event::Data => {
-                            // Enable FIFO request interrupt
-                            self.rb.cha.im.modify(|_, w| w.freqie().set_bit());
-                        }
+                        Event::Overdue              => ch.im.write(|w| w.ovrudrie().clear_bit()),
+                        Event::Muted                => ch.im.write(|w| w.mutedetie().clear_bit()),
+                        Event::WrongClock           => ch.im.write(|w| w.wckcfgie().clear_bit()),
+                        Event::Data                 => ch.im.write(|w| w.freqie().clear_bit()),
+                        Event::AnticipatedFrameSync => ch.im.write(|w| w.afsdetie().clear_bit()),
+                        Event::LateFrameSync        => ch.im.write(|w| w.lfsdetie().clear_bit()),
                     }
                 }
 
-                /// Stop listening for `event`
-                pub fn unlisten(&mut self, event: Event) {
+                /// Clears interrupt flag `event` on the `channel`
+                ///
+                /// Note: Event::Data is accepted but does nothing as that flag is cleared by reading/writing data
+                pub fn clear_irq(&mut self, channel: SaiChannel, event: Event) {
+                    let ch = match channel {
+                        SaiChannel::ChannelA => &self.rb.cha,
+                        SaiChannel::ChannelB => &self.rb.chb,
+                    };
                     match event {
-                        Event::Data => {
-                            // Enable FIFO request interrupt
-                            self.rb.cha.im.modify(|_, w| w.freqie().clear_bit());
-                        }
+                        Event::Overdue              => ch.clrfr.write(|w| w.covrudr().set_bit()),
+                        Event::Muted                => ch.clrfr.write(|w| w.cmutedet().set_bit()),
+                        Event::WrongClock           => ch.clrfr.write(|w| w.cwckcfg().set_bit()),
+                        Event::Data                 => (), // Cleared by reading/writing data
+                        Event::AnticipatedFrameSync => ch.clrfr.write(|w| w.cafsdet().set_bit()),
+                        Event::LateFrameSync        => ch.clrfr.write(|w| w.clfsdet().set_bit()),
                     }
                 }
 
-                /// Clears interrupt flag
-                pub fn clear_irq(&mut self) {
-                    self.rb.cha.clrfr.write(|w| {
-                        // FIFO request interrupt is cleared by
-                        // reading or writing data
-                        w
-                    });
+                /// Clears all interrupts on the `channel`
+                pub fn clear_all_irq(&mut self, channel: SaiChannel) {
+                    match channel {
+                        SaiChannel::ChannelA => &self.rb.cha.clrfr.reset(),
+                        SaiChannel::ChannelB => &self.rb.chb.clrfr.reset(),
```

Comment:
This doesn't quite do what you expect. The `reset` method ([from svd2rust](https://docs.rs/svd2rust/0.17.0/svd2rust/)) writes the reset value as defined in the SVD to the register. In this case the reset value is all zeros, which doesn't clear any of the flags.